src/main/java/org/spongepowered/common/data/processor/value/entity/IsFlyingValueProcessor.java

```diff
@@ -0,0 +1,114 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.processor.value.entity;
+
+import com.google.common.base.Optional;
+import net.minecraft.entity.player.EntityPlayer;
+import org.spongepowered.api.data.DataTransactionBuilder;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.value.BaseValue;
+import org.spongepowered.api.data.value.ValueContainer;
+import org.spongepowered.api.data.value.immutable.ImmutableValue;
+import org.spongepowered.api.data.value.mutable.Value;
+import org.spongepowered.common.data.processor.common.AbstractSpongeValueProcessor;
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeValue;
+import org.spongepowered.common.data.value.mutable.SpongeValue;
+
+public class IsFlyingValueProcessor extends AbstractSpongeValueProcessor<Boolean, Value<Boolean>> {
+
+    protected IsFlyingValueProcessor() {
+        super(Keys.IS_FLYING);
+    }
+
+    @Override
+    public Optional<Boolean> getValueFromContainer(ValueContainer<?> container) {
+        if (supports(container)) {
+            net.minecraft.entity.Entity entity = ((net.minecraft.entity.Entity) container);
+            boolean isFlying = entity.isAirBorne;
+            return Optional.of(isFlying);
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public Optional<Value<Boolean>> getApiValueFromContainer(ValueContainer<?> container) {
+        if (supports(container)) {
+            net.minecraft.entity.Entity entity = ((net.minecraft.entity.Entity) container);
+            boolean isFlying = entity.isAirBorne;
+            return Optional.<Value<Boolean>>of(new SpongeValue<Boolean>(Keys.IS_FLYING, isFlying));
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public boolean supports(ValueContainer<?> container) {
+        return container instanceof net.minecraft.entity.Entity;
+    }
+
+    @SuppressWarnings("unchecked")
+    public DataTransactionResult offerToStore(ValueContainer<?> container, BaseValue<?> value) {
```

Comment:
This shouldn't exist. Use `offerToStore(ValueContainer<?> container, Boolean value)`.